### DISCUSSION TITLE
Changing container name.

### DISCUSSION THREAD

**Discussion by wawansoer** [coollabsio/coolify#3231] [Q&A]
Hi guys 
I would like to ask, is it possible to change the name of the container that Coolify has created? Is there any effect if this container name is changed?

Sorry for a dump question

**Comment by SalyczeQ** [comment]
Yes, you can setup the container name for resource statically. But you will lose rolling update feature.
You can find it in resource setting: Configuration > Advanced > Container Names

**Reply by VincenzoKoestler** [reply]
This seems not to be possible for service stacks? I tried adding the "COOLIFY_CONTAINER_NAME" ENV but it still deploys using its own name?

**Reply by tylerdurden4285** [reply]
Maybe it's a dumb question but why does changing the name to something custom remove rollback ability?

**Reply by Cinzya** [reply]
@tylerdurden4285 When you redeploy your app / service. Coolify will spin up a new container and then route traffic to the newly created container after it finished building and is ready to receive traffic.

This means for a short moment, two containers exist for the same app / service. Since two containers can't share the exact name, coolify needs to give them different names, in form of a UUID. So if you enable persistent container names, that means coolify needs to stop and remove the old container before it can create a new one.

**Reply by tylerdurden4285** [reply]
@Cinzya I just wanted to say thank you for taking the time to write a very clear and concise response that answered the question directly. I appreciate it. Coolify is growing so well. Thanks!

**Reply by oribenez** [reply]
> @tylerdurden4285 When you redeploy your app / service. Coolify will spin up a new container and then route traffic to the newly created container after it finished building and is ready to receive traffic.
> 
> This means for a short moment, two containers exist for the same app / service. Since two containers can't share the exact name, coolify needs to give them different names, in form of a UUID. So if you enable persistent container names, that means coolify needs to stop and remove the old container before it can create a new one.

so is there a way to add a prefix name to the container?
in this way we can still have the rolling updates

**Reply by nelsonmestevao** [reply]
I am looking for something like @oribenez mentions. However, it seems like that possibility does not exist yet?

**Reply by humblemodulo** [reply]
Running `v4.0.0-beta.418` and it appears that `Configuration > Advanced > Container Names` no longer exists.  😖  We really need the ability to work with "guaranteed" names -- even if we must silo via a separate "foot-gun" docker configuration.

Obviously the manual option exists, but that has it's own issues.  🫤

Still, thanks for letting us know "why"?

**Comment by watate** [comment]
I'd be interested to know the solution to this as well! I monitor my containers' resource usage with Glances, and it's pretty hard to tell which app is using what resource when it's just a random string like: d404c4wc0wck40gskgc4wg4k

**Reply by nelsonmestevao** [reply]
Yes, this is the main reason why I want a prefix name for the container :+1:

**Comment by Killian-Aidalinfo** [comment]
Hi, go to advanced and enable Raw Compose Deployment. Please note that if you use Coolify's automatic proxy configuration, it will no longer work.

**Comment by Cinzya** [comment]
Just sharing for awareness, because there are multiple discussion around this topic but they are only mentioned briefly in threads. I would recommend upvoting this discussion, if people are looking for a prefix container name feature: https://github.com/coollabsio/coolify/discussions/3569

**Comment by helderpgoncalves** [comment]
I am looking forward to see if anyone grabs and discuss better this topic. It's an absolutely nightmare to monitor containers when the names are random.

**Comment by Burserk84** [comment]
you cango to your applcation then check the vox use raw conrainer and it will use the name that you have defined

**Comment by Burserk84** [comment]
you cango to your applcation > advanced > Raw Compose Deployment
 and it will use the name that you have defined
